### ISSUE TITLE
Make error logs so one can click into file when possible

### ISSUE THREAD

**Issue by bep** [gohugoio/hugo#5332]
https://discourse.gohugo.io/t/help-test-browser-errors-etc-new-in-next-hugo-0-50/14745/8?u=bep

**Comment by bep** [comment]
>So would it be possible to make this configurable?

That would be a future request. As usual, I have taken on too much already ... But it's going to be nice ...

**Comment by bep** [comment]
@kaushalmodi thinking about it, a configuration option for this sounds like little work compared to the benefit. How would you suggest we make that configuration look like?

**Comment by kaushalmodi** [comment]
To avoid maintenance annoyance for a variety of editors out there, I'd suggest to have a single map-style config variable in the site's config.toml.

```toml
[Editor]
  emacs = "+LINE:COL FILE"
  vim = "+LINE FILE"
  sublime = "FILE:LINE"
```

*`LINE`, `COL`, `FILE` are special placeholders.*


How this will work:

1. Internally, `hugo` will check the `$EDITOR` environment variable. Let's say it finds that to be `emacsclient`.
2. Looping through all the keys in the `.Site.Editor` map, it will use the first key which matches with the **beginning** of `$EDITOR`. In above example, as the env var is set to `emacsclient`, the `emacs` key would match.
3. Then simply replace the LINE, COL and FILE placeholders with the actual values in the value of that selected key. Let's say that value is a string "foo". Then `hugo` simply launches `$EDITOR foo`.
4. If the `$EDITOR` environment variable is **not** set, don't hyperlink the line in the error message.

**Comment by bep** [comment]
This is fixed, see #5352 for the config bit.
